src/main/java/com/amihaiemil/docker/RtContainer.java

```diff
@@ -38,7 +44,32 @@
 
     @Override
     public JsonObject inspect() throws IOException {
-        return null;
+        final HttpGet inspect = new HttpGet(this.baseUri.toString() + "/json");
+        final HttpResponse response = this.client.execute(inspect);
+        final JsonObject info;
+        if(response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+            info = Json
+                .createReader(response.getEntity().getContent()).readObject();
+        } else {
+            info = null;
+        }
+        inspect.releaseConnection();
+        return info;
+    }
+
+    @Override
+    public void start() throws IOException {
+        final HttpPost start = new HttpPost(
+            this.baseUri.toString() + "/start"
+        );
+        final HttpResponse response = this.client.execute(start);
+        final int status = response.getStatusLine().getStatusCode();
+        if(status!= HttpStatus.SC_NO_CONTENT) {
+            throw new IllegalStateException(
```

Comment:
@amihaiemil why not throw `IOException`?

Reply:
@llorllale Because, as I understand it, ``IOException`` should be used only when there are real issues on the network. We have a puzzle for that, we would need an ``UnexpectedResponseException``, which would provide the status and reason phrase to the user. 

So, once that puzzle is fixed, this throw will be changed as well, it won't be ``IllegalStateException``.

If we simply throw ``IOException`` the user won't know whether there was a real networking issue (e.g. timeout connection) or something else.

What do you think?

Reply:
@amihaiemil It's a design decision. We are currently not declaring any other checked exception, so I assume this new `UnexpectedResponseException` would be runtime? I'd throw an `IOException` (or a subclass) because it's already in the API's contract. 

I think this is a design call and that you have final say.